charts/seed-bootstrap/prometheus-rules/recording-rules.rules.yaml

```diff
@@ -36,6 +36,110 @@ groups:
   - record: seed:hvpa_status_applied_vpa_recommendation_target:sum_by_container
     expr: sum(hvpa_status_applied_vpa_recommendation{recommendation="target"}) by (container, resource)
 
+  # Recording rules for the count of containers which have CPU usage at or above 80% of their request, aggregated by containers.
+  - record: seed:containers_over_cpu_scaling_threshold_total:count_by_container
+    expr: count(max(rate(container_cpu_usage_seconds_total[5m])) by (container, namespace) / max(kube_pod_container_resource_requests_cpu_cores) by (container, namespace) >= 0.8) by (container)
+
+  # Recording rules for the count of containers which have memory usage at or above 120% of their request, aggregated by containers.
+  - record: seed:containers_over_memory_scaling_threshold_total:count_by_container
+    expr: count(max(container_memory_working_set_bytes) by (container, namespace) / max(kube_pod_container_resource_requests_memory_bytes) by (container, namespace) >= 1.2) by (container)
+
+  # Recording rules for the count of containers which have CPU usage at or above 80% of their limits, aggregated by containers.
+  - record: seed:containers_over_cpu_limiting_threshold_total:count_by_container
```

Comment:
No. The expressions are different.

`seed:containers_over_cpu_scaling_threshold_total:count_by_container` is based on `kube_pod_container_resource_requests_cpu_cores` indicating how many namespaces have this container at or above the threshold where they might be scaled. 0.8 = 80% because that is the utilization percentage when HPA is suppose scale the replicas. This is useful to evaluate how many shoot namespaces are at threshold of HPA scaling.

`seed:containers_over_cpu_limiting_threshold_total:count_by_container` is based on indicating how many namespaces have this container at or above the threshold where their resources might be limited by docker/kubelet. This can help in analysing how many shoots are at risk of being affected if scaling (horizontally or vertically) is blocked due to any reason (bug, deliberate or wrong configuration).

Reply:
Ah ok my bad. I should have compared them side by side. Then the metrics are fine.